### PR TITLE
fix: Refactor integrator ID and add Prometheus metrics

### PR DESCRIPTION
**Pull request by PirosB3**
1. Adds more Prometheus counters in the networking layer of HTTP(s) requests towards market makers
2. Adds a little more documentation and structure in the type formatting
3. Changes naming from `apiKey` to `integrator`

### CODE REVIEW

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 1]
```diff
@@ -472,18 +486,28 @@ export class QuoteRequestor {
                 try {
                     if (typedMakerUrl.pairType === RfqPairType.Standard) {
                         const response = await this._quoteRequestorHttpClient.get(`${typedMakerUrl.url}/${quotePath}`, {
-                            headers: { '0x-api-key': options.apiKey },
+                            headers: { '0x-api-key': options.integrator.integratorId },
```
Comment (phil-ociraptor) [inline packages/asset-swapper/src/utils/quote_requestor.ts:489]: Can we send this value twice, as the 0x-api-key header as well as 0x-integrator-id? That way we can one day remove the concept of 0x-api-key for quote served

**Review by phil-ociraptor — APPROVED** [review]
Looks good - we could even remove integrator id check from asset swapper, but I think this is good!

File: packages/asset-swapper/src/swap_quoter.ts [hunk 2]
```diff
@@ -414,12 +418,11 @@ export class SwapQuoter {
         return isOpenOrder && !willOrderExpire && isFeeTypeAllowed;
     }; // tslint:disable-line:semicolon
 
-    private _isApiKeyWhitelisted(apiKey: string | undefined): boolean {
+    private _isIntegrationKeyWhitelisted(apiKey: string | undefined): boolean {
```
Comment (rhinodavid) [inline packages/asset-swapper/src/swap_quoter.ts:421]:
```suggestion
    private _isIntegratorIdWhitelisted(integratorId: string | undefined): boolean {
```

File: packages/asset-swapper/src/swap_quoter.ts [hunk 3]
```diff
@@ -438,19 +441,19 @@ export class SwapQuoter {
...
+        const { integrator, nativeExclusivelyRFQ, intentOnFilling, txOrigin } = rfqt;
         // If RFQ-T is enabled and `nativeExclusivelyRFQ` is set, then `ERC20BridgeSource.Native` should
         // never be excluded.
         if (nativeExclusivelyRFQ === true && !sourceFilters.isAllowed(ERC20BridgeSource.Native)) {
             throw new Error('Native liquidity cannot be excluded if "rfqt.nativeExclusivelyRFQ" is set');
         }
 
         // If an API key was provided, but the key is not whitelisted, raise a warning and disable RFQ
-        if (!this._isApiKeyWhitelisted(apiKey)) {
+        if (!this._isIntegrationKeyWhitelisted(integrator.integratorId)) {
```
Comment (rhinodavid) [inline packages/asset-swapper/src/swap_quoter.ts:452]:
```suggestion
        if (!this._isIntegratorIdWhitelisted(integrator.integratorId)) {
```

File: packages/asset-swapper/src/swap_quoter.ts [hunk 4]
```diff
@@ -438,19 +441,19 @@ export class SwapQuoter {
...
         // tslint:disable-next-line: boolean-naming
-        const { apiKey, nativeExclusivelyRFQ, intentOnFilling, txOrigin } = rfqt;
+        const { integrator, nativeExclusivelyRFQ, intentOnFilling, txOrigin } = rfqt;
         // If RFQ-T is enabled and `nativeExclusivelyRFQ` is set, then `ERC20BridgeSource.Native` should
         // never be excluded.
         if (nativeExclusivelyRFQ === true && !sourceFilters.isAllowed(ERC20BridgeSource.Native)) {
             throw new Error('Native liquidity cannot be excluded if "rfqt.nativeExclusivelyRFQ" is set');
         }
 
         // If an API key was provided, but the key is not whitelisted, raise a warning and disable RFQ
```
Comment (rhinodavid) [inline packages/asset-swapper/src/swap_quoter.ts:451]:
```suggestion
        // If an integrator ID was provided, but the ID is not whitelisted, raise a warning and disable RFQ
```

File: packages/asset-swapper/src/types.ts [hunk 5]
```diff
@@ -243,7 +243,7 @@ export interface RfqmRequestOptions extends RfqRequestOpts {
 export interface RfqRequestOpts {
     takerAddress: string;
     txOrigin: string;
-    apiKey: string;
+    integrator: Integrator
```
Comment (rhinodavid) [inline packages/asset-swapper/src/types.ts:246]: this is nice

File: packages/asset-swapper/src/types.ts [hunk 6]
```diff
@@ -294,8 +294,13 @@ export interface RfqFirmQuoteValidator {
     getRfqtTakerFillableAmountsAsync(quotes: RfqOrder[]): Promise<BigNumber[]>;
 }
 
+export interface Integrator {
```
Comment (rhinodavid) [inline packages/asset-swapper/src/types.ts:297]: could we also put the URI whitelist here? Not sure if that makes sense

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 7]
```diff
@@ -61,6 +61,20 @@ export interface MetricsProxy {
      * @param expirationTimeSeconds the expiration time in seconds
      */
     incrementFillRatioWarningCounter(isLastLook: boolean, maker: string): void;
+
+
+    logRfqMakerInteraction(interaction: {
+        isLastLook: boolean,
+        integrator: {
```
Comment (rhinodavid) [inline packages/asset-swapper/src/utils/quote_requestor.ts:68]: can we do `integrator: Integrator`?

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 1 above)
Comment (rhinodavid) [reply packages/asset-swapper/src/utils/quote_requestor.ts:489]: love it

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 8]
```diff
@@ -549,7 +573,7 @@ export class QuoteRequestor {
                     rfqMakerBlacklist.logTimeoutOrLackThereof(typedMakerUrl.url, latencyMs >= timeoutMs);
                     this._warningLogger(
                         convertIfAxiosError(err),
-                        `Failed to get RFQ-T ${quoteType} quote from market maker endpoint ${typedMakerUrl.url} for API key ${options.apiKey} for taker address ${options.takerAddress} and tx origin ${options.txOrigin}`,
+                        `Failed to get RFQ-T ${quoteType} quote from market maker endpoint ${typedMakerUrl.url} for API key ${options.integrator.integratorId} (${options.integrator.label}) for taker address ${options.takerAddress} and tx origin ${options.txOrigin}`,
```
Comment (rhinodavid) [inline packages/asset-swapper/src/utils/quote_requestor.ts:576]:
```suggestion
                        `Failed to get RFQ-T ${quoteType} quote from market maker endpoint ${typedMakerUrl.url} for integrator ${options.integrator.integratorId} (${options.integrator.label}) for taker address ${options.takerAddress} and tx origin ${options.txOrigin}`,
```

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 1 above)
Comment (PirosB3) [reply packages/asset-swapper/src/utils/quote_requestor.ts:489]: Mindblown. Thank you

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 7 above)
Comment (PirosB3) [reply packages/asset-swapper/src/utils/quote_requestor.ts:68]: Sure

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 1 above)
Comment (PirosB3) [reply packages/asset-swapper/src/utils/quote_requestor.ts:496]: Just thinking about this a sec, @phil-ociraptor - if we create custom API keys that can only connect to a subset of services. Won't we risk doxxing integrators if there is some bug?

Example: we create a "Jump Trading" API key, which whitelists only Jump. A regression bug happens and suddently Wintermute can see requests from Jump Trading?

**Review by rhinodavid — APPROVED** [review]

**Pull request merged**